src/core/insights/command-queue.ts

```diff
@@ -0,0 +1,137 @@
+        return (...args: unknown[]) => {
+          if (!target.isInitialized || !target.realImplementation) {
+            // Queue the command for later execution
+            if (prop === "initInsights") {
+              // Special handling for initInsights - execute it right away
+              target.executeInitInsights(args[0] as InsightsConfig);
+              return;
+            }
+
+            // For debug logging
+            if (target.debugMode) {
+              logger.debug(`Queuing method call: ${String(prop)}`, args);
+            }
+
+            target.queue.push({
+              methodName: prop as keyof AnalyticsService,
+              args,
+            });
+            return;
+          }
+
+          // Execute the command immediately on the real implementation
+          if (
+            typeof (target.realImplementation as AnalyticsService)[prop] ===
+            "function"
+          ) {
+            return (target.realImplementation as AnalyticsService)[prop](
+              ...args,
+            );
+          }
+        };
+      },
+    });
+  }
+
+  // Special handling for init since it needs to happen right away
+  private executeInitInsights(config: InsightsConfig): void {
+    this.debugMode = !!config.debug;
+
+    if (this.debugMode) {
+      logger.debug("Initializing insights");
+    }
+
+    // Create and initialize the real implementation
+    this.realImplementation = new InsightsService();
+    this.realImplementation.initInsights(config);
+
+    // Mark as initialized and process the queue
+    this.isInitialized = true;
+    this.processQueue();
+  }
+
+  // Process all queued commands
+  private processQueue(): void {
+    if (this.debugMode) {
+      logger.debug(`Processing ${this.queue.length} queued commands`);
+    }
+
+    while (this.queue.length > 0) {
+      const cmd = this.queue.shift();
+      if (
+        cmd &&
+        this.realImplementation &&
+        typeof this.realImplementation[cmd.methodName] === "function"
+      ) {
+        try {
+          if (this.debugMode) {
+            logger.debug(
+              `Executing queued command: ${cmd.methodName}`,
+              cmd.args,
+            );
+          }
+          (
+            this.realImplementation[cmd.methodName] as (
+              ...args: unknown[]
+            ) => unknown
+          )(...(cmd.args as unknown[]));
+        } catch (e) {
+          if (this.debugMode) {
+            logger.error(
+              `Error executing queued command: ${cmd.methodName}`,
+              e,
+            );
+          }
+        }
+      }
+    }
+  }
+
+  // Implement all methods required by AnalyticsService to satisfy TypeScript
+  // (These won't be called directly due to the proxy)
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  initInsights(config: InsightsConfig): void {}
+  enableDebugMode(): void {}
+  disableDebugMode(): void {}
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
Consider the following, in `eslintrc.cjs`, add the following under `rules` and prefix the arguments with underscores. Works for me
```ts
"@typescript-eslint/no-unused-vars": [
  "error",
  {
    argsIgnorePattern: "^_",
    varsIgnorePattern: "^_",
  },
],
```